vaadin-virtual-list-flow-parent/vaadin-virtual-list-flow/src/main/resources/META-INF/resources/frontend/virtualListConnector.js

```diff
@@ -49,31 +49,41 @@ window.Vaadin.Flow.virtualListConnector = {
 
     requestAnimationFrame(() => updateRequestedItem);
 
-    /*
-     * Ensure all items that virtual list will be looking at are actually defined.
-     * If this is not done, the component will keep looking ahead through the
-     * array until finding enough present items to render. In our case, that's
-     * a really slow way of achieving nothing since the rest of the array is
-     * empty.
-     */
-    const originalRenderer = list.renderer;
-    list.renderer = function (root, list, model) {
-      root.__virtualListIndex = model.index;
-
-      if (model.item === undefined) {
-        originalRenderer.call(list, root, list, {...model, item: list.$connector.placeholderItem});
-      } else {
-        originalRenderer.call(list, root, list, model);
+    // Add an observer function that will invoke on virtualList.renderer property
+    // change and then patches it with a wrapper renderer
+    list.patchVirtualListRenderer = function () {
```

Comment:
Needed to change this logic a bit.

The previous revision assumed the `virtual-list` already has a `renderer` function at this point and patched it. That may not always be the case. For example, a `LitRenderer` or a `<template>` tag may be added dynamically. In that case, the connector would end up patching a non-existing function and also the patched function could get overridden later.

The new version observes for changes in `virtual-list.renderer` and patches the function when it's assigned.